src/sample_pool.rs

```diff
@@ -905,7 +905,10 @@ fn main() {
 
 								if leading_zeros >= client_target + WEAK_BLOCK_RATIO_0S {
 									if client.submitted_header_hashes.try_insert(&sketch.header_prevblock, block_hash) {
-										weak_block_submitted(&*submitter_state, client_id, &sketch.user_tag_1, our_payout, &header, &new_txn, &sketch.extra_block_data, leading_zeros, client_target);
+										let block_target = utils::nbits_to_target(sketch.header_nbits);
+										let is_good_block = utils::does_hash_meet_target(&block_hash[..], &block_target[..]);
```

Comment:
Note that this is unverified - a user can claim whatever they want for nBits and it's unchecked (at least for now - this will need to change pre-production but I'm not sure how yet - whether we want to call weak_block_submitted before its valiadted, or after, in which case payout needs to come based on the duplicative non-weak-block share submission).

Reply:
So now we checked prev_block_hash, coinbase_tx and if share/weak_block hash is duplicate. If a client want to claim a false nBits or n, does it